xarray/core/indexing.py

```diff
@@ -820,7 +839,7 @@ def explicit_indexing_adapter(
     result = raw_indexing_method(raw_key.tuple)
     if numpy_indices.tuple:
         # index the loaded np.ndarray
-        result = NumpyIndexingAdapter(np.asarray(result))[numpy_indices]
+        result = NumpyIndexingAdapter(result)[numpy_indices]
```

Comment:
Not entirely sure about this. Only the pydap test failed, so I converted that to np.array in the backend.